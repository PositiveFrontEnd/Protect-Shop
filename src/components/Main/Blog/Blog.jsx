import React from "react";
import "./Blog.scss"
import SectionBlogImgOne from "/Images/sectionBlogOne.jpg"
import SectionBlogTwo from "/Images/sectionBlogTwo.jpg"
import SectionBlogThree from "/Images/sectionBlogThree.jpg"
import Button from "../../Button/Button"
import { useNavigate } from "react-router-dom";

const Blog = () => {
    const navigate = useNavigate()
    const navigateBlogPage = () => {
        navigate('/blog')
    }
    return (
        <>
            <section className="blog__wrapper">
                <div className="container">
                    <h2 className="blog__title">blog</h2>
                    <ul className="blog__info">
                        <li className="mobile__description">
                            <img className="blog__img" src={SectionBlogImgOne} alt="sectionBlogOne" />
                            <h4 className="blog__info__title">Mini bags forever </h4>
                            <p className="blog__info__description">Small bags have become very popular this season.
                                They are perfect for parties and weekends when you don't have to wear many things.
                                Such bags can be worn both on the shoulder and at the fingertips.</p>
                        </li>
                        <li className="develop__description">
                            <img className="blog__img" src={SectionBlogTwo} alt="" />
                            <h4 className="blog__info__title">Mini bags forever </h4>
                            <p className="blog__info__description">Small bags have become very popular this season.
                                They are perfect for parties and weekends when you don't have to wear many things.
                                Such bags can be worn both on the shoulder and at the fingertips.</p>
                        </li>
                        <li className="develop__description">
                            <img className="blog__img" src={SectionBlogThree} alt="" />
                            <h4 className="blog__info__title">Sackbags</h4>
                            <p className="blog__info__description">Sackbags with large fasteners and chains have become popular this season.</p>
                            <p className="blog__info__description">These bags are suitable for many styles, from classics to modern minimalism.</p>
                        </li>
                    </ul>
                    <div className="blog__button" >
                        <Button click={navigateBlogPage} white>See more</Button>
                    </div>
                </div>
            </section>
        </>
    )


}

export default Blog