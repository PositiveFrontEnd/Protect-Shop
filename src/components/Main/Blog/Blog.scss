.blog__wrapper {
  background: #ebebe9;

  .blog__title {
    margin-bottom: 12px;
    padding-top: 24px;
    font-size: 18px;
    font-family: "Noto Sans", sans-serif;
    text-align: center;
    text-transform: uppercase;

    @media (width >= 768px) {
      margin-bottom: 60px;
      padding-top: 60px;
      font-size: 24px;
    }
  }

  .blog__info {
    display: flex;

    @media (width >= 768px) {
      gap: 12px;
    }
  }

  .blog__info__title {
    margin-top: 16px;
    margin-bottom: 8px;
    color: var(--black);
    font-size: 14px;
    text-transform: uppercase;
  }

  .blog__info__description {
    max-width: 634px;
    font-size: 14px;
  }

  .blog__img {
    width: 100%;
  }

  .blog__button {
    display: flex;
    justify-content: center;
    padding-top: 27px;
    padding-bottom: 24px;

    @media (width >= 768px) {
      padding-bottom: 60px;
    }
  }

  .mobile__description {
    @media (width >= 768px) {
      display: none;
    }
  }

  .develop__description {
    display: none;

    @media (width >= 768px) {
      display: block;
    }
  }
}
