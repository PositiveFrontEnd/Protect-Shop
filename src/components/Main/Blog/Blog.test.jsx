import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import Blog from "./Blog";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  Blog", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {

        };

        const store = mockStore(initialState);
        const blog = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Blog />
                </BrowserRouter>
            </Provider>
        );

        expect(blog).toMatchSnapshot();
    });
});
