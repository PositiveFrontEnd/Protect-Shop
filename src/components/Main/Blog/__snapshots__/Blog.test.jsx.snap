// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  Blog > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="blog__wrapper"
      >
        <div
          class="container"
        >
          <h2
            class="blog__title"
          >
            blog
          </h2>
          <ul
            class="blog__info"
          >
            <li
              class="mobile__description"
            >
              <img
                alt="sectionBlogOne"
                class="blog__img"
                src="/Images/sectionBlogOne.jpg"
              />
              <h4
                class="blog__info__title"
              >
                Mini bags forever 
              </h4>
              <p
                class="blog__info__description"
              >
                Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
              </p>
            </li>
            <li
              class="develop__description"
            >
              <img
                alt=""
                class="blog__img"
                src="/Images/sectionBlogTwo.jpg"
              />
              <h4
                class="blog__info__title"
              >
                Mini bags forever 
              </h4>
              <p
                class="blog__info__description"
              >
                Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
              </p>
            </li>
            <li
              class="develop__description"
            >
              <img
                alt=""
                class="blog__img"
                src="/Images/sectionBlogThree.jpg"
              />
              <h4
                class="blog__info__title"
              >
                Sackbags
              </h4>
              <p
                class="blog__info__description"
              >
                Sackbags with large fasteners and chains have become popular this season.
              </p>
              <p
                class="blog__info__description"
              >
                These bags are suitable for many styles, from classics to modern minimalism.
              </p>
            </li>
          </ul>
          <div
            class="blog__button"
          >
            <button
              class="button-white"
              type="button"
            >
              See more
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="blog__wrapper"
    >
      <div
        class="container"
      >
        <h2
          class="blog__title"
        >
          blog
        </h2>
        <ul
          class="blog__info"
        >
          <li
            class="mobile__description"
          >
            <img
              alt="sectionBlogOne"
              class="blog__img"
              src="/Images/sectionBlogOne.jpg"
            />
            <h4
              class="blog__info__title"
            >
              Mini bags forever 
            </h4>
            <p
              class="blog__info__description"
            >
              Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
            </p>
          </li>
          <li
            class="develop__description"
          >
            <img
              alt=""
              class="blog__img"
              src="/Images/sectionBlogTwo.jpg"
            />
            <h4
              class="blog__info__title"
            >
              Mini bags forever 
            </h4>
            <p
              class="blog__info__description"
            >
              Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
            </p>
          </li>
          <li
            class="develop__description"
          >
            <img
              alt=""
              class="blog__img"
              src="/Images/sectionBlogThree.jpg"
            />
            <h4
              class="blog__info__title"
            >
              Sackbags
            </h4>
            <p
              class="blog__info__description"
            >
              Sackbags with large fasteners and chains have become popular this season.
            </p>
            <p
              class="blog__info__description"
            >
              These bags are suitable for many styles, from classics to modern minimalism.
            </p>
          </li>
        </ul>
        <div
          class="blog__button"
        >
          <button
            class="button-white"
            type="button"
          >
            See more
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
